cpmpy/expressions/core.py

```diff
@@ -465,6 +468,10 @@ def _wsum_make(arg):
         return arg.args
     elif arg.name == 'mul':
         return [arg.args[0]], [arg.args[1]]
+    elif arg.name == "-" and isinstance(arg.args[0], Operator):
+        # - (3 * y)
+        w, x = _wsum_make(arg.args[0]) 
```

Comment:
that looks very broad catching... so for any operator, we forably turn it into a wsum and make the 'w' negative?

is there not a more specific case you want to catch?

maybe at least a wsum_shuld on args0?


or is it the recursive case you want to catch; e.g. a wsum in a wsum? I'm again not sure we should do this her ein 'expression', but perhaps rather in flatten...?

Reply:
Changed the rewriting rules for - in mul, wsum, .... to flattening.